### PR TITLE
Adding custom error message

### PR DESCRIPTION
**Pull request by rek**
This allows the use of custom error messages for validation errors.

### CODE REVIEW

**Comment by stuyam** [comment]
@rek this is an awesome contribution thank you! Since I merged your last PR there are merge conflicts on this current PR. Would you mind resolving the conflicts before I merge?

**Comment by rek** [comment]
Yup sure. I didn't wanna send them together just in case you didn't wanna merge one :)

PS. awesome lib, I love the simplicity.

File: src/simple-react-validator.js [hunk 1]
```diff
@@ -65,7 +65,7 @@ class SimpleReactValidator{
         this.fields[field] = false;
         if(this.messagesShown){
           var re = /_/g;
-          message = this.rules[rule].message.replace(':attribute', field.replace(re, ' '));
+          message = customErrors[rule] || this.rules[rule].message.replace(':attribute', field.replace('_', ' '));
```
Comment (stuyam) [inline src/simple-react-validator.js]: @rek should't this be using the `re` var?

Comment (stuyam) [inline src/simple-react-validator.js]: just update master to fix a few things including this 👍

Comment (rek) [inline src/simple-react-validator.js]: Whoops thanks, sorry I missed that!

**Comment by stuyam** [comment]
@rek are you using simple-react-validator on project you are working on? Any suggestions or feedback about the project?

**Comment by rek** [comment]
Yup we are testing it out on a project, I choose it because of how simple and lightweight it was. I like how easy it was to implement into my forms.

The only other nice feature I would like is the ability to wrap the original input, so we can have a class change on that for errors, but to do that, you would have to pass the input into `validator.message`, though that might be quite a significant change.

**Comment by stuyam** [comment]
Yeah it might change how the main components of the lib work. I am not against it though. But that gets into like breaking a form validator component out into it's own thing maybe which I am not 100% against but I am trying to keep the moving parts to a minimum.

Maybe there is a way we could achieve both. I am open to suggestions on how something like that might work.

**Comment by rek** [comment]
Yer I am not sure, perhaps I can put a proof of concept together we can look at.

Though I too am a little bit against adding too many moving parts.

**Comment by rek** [comment]
Something like this seems quite simple: https://github.com/rek/simple-react-validator/commit/047b9b18c7b88fa3fc70deed8336e9bb52bf9af6

Though would need some changes to be a bit more functional, and that list of args is getting quite long, would almost be at the state where it would be better as an object.

**Pull request merged**